compiler-core/src/exhaustiveness.rs

```diff
@@ -3558,6 +3575,92 @@ fn segment_matched_value(
     }
 }
 
+fn int_to_bits(
+    value: &BigInt,
+    read_size: &ReadSize,
+    endianness: Endianness,
+    signed: bool,
+) -> Result<BitVec<u8, Msb0>, IntToBitsError> {
+    let size = read_size
+        .constant_bits()
+        .ok_or(IntToBitsError::NonConstantSize)?
+        .to_u32()
+        .ok_or(IntToBitsError::ExceedsMaximumSize)?;
+
+    if !representable_with_bits(value, size, signed) {
+        return Err(IntToBitsError::Unrepresentable { size });
+    } else if size > BitArrayMatchedValue::MAX_BITS_INTERFERENCE {
+        return Err(IntToBitsError::ExceedsMaximumSize);
+    }
+
+    // Pad negative numbers with 1s (true) and non-negative numbers with 0s (false)
+    let pad_digit = value.sign() == Sign::Minus;
+    let size = size as usize;
+    let mut bytes = int_to_bytes(value, endianness, signed);
+    let bytes_size = bytes.len() * 8;
+
+    // There are 3 cases, which are easier to handle separately by endianness
+    // If the size of the bigint bytes equals the expected bits, we can return them as-is
+    // If there are more bits than we need, we need to trim some of the most significant bits.
+    // E.g. `6:3` yields one byte of which we need to trim the 5 most significant bits.
+    // Values like 999:3 are illegal and caught by the guard at the start of the function.
+    // If there are fewer bits than we need, we need to add some.
+    // E.g. `6:13` yields one byte which we need to pad with 5 bits
+    let bits = match (endianness, bytes_size.cmp(&size)) {
+        (_, Ordering::Equal) => BitVec::from_vec(bytes),
+
+        (Endianness::Big, Ordering::Greater) => {
+            BitVec::from_bitslice(&bytes.view_bits()[bytes_size - size..])
+        }
```

Comment:
Maybe I'm not understanding this properly but shouldn't it be impossible to have a `Greater` here given the previous checks?